x/concentrated-liquidity/internal/math/tick.go

```diff
@@ -2,46 +2,123 @@ package math
+	priceUpperTick, err := TickToPrice(sdk.NewInt(upperTick), kAtPriceOne)
 	if err != nil {
 		return sdk.Dec{}, sdk.Dec{}, err
 	}
-	sqrtPriceLowerTick, err := TickToSqrtPrice(sdk.NewInt(lowerTick))
+	priceLowerTick, err := TickToPrice(sdk.NewInt(lowerTick), kAtPriceOne)
 	if err != nil {
 		return sdk.Dec{}, sdk.Dec{}, err
 	}
-	return sqrtPriceLowerTick, sqrtPriceUpperTick, nil
+	return priceLowerTick, priceUpperTick, nil
 }
 
-// TickToSqrtPrice takes the tick index and returns the corresponding sqrt of the price.
-// Returns error if fails to calculate sqrt price. Otherwise, the computed value and nil.
-// TODO: test
-func TickToSqrtPrice(tickIndex sdk.Int) (sqrtPrice sdk.Dec, err error) {
-	// If the tick index is positive, we can use the normal equation to calculate the square root price.
-	// However, if the tick index is negative, since we cannot take negative powers with the sdk,
-	// we need to take one over the original equation in order to make the power positive.
-	if tickIndex.GTE(sdk.ZeroInt()) {
-		sqrtPrice, err = sdk.NewDecWithPrec(10001, 4).Power(tickIndex.Uint64()).ApproxSqrt()
+// TickToSqrtPrice calculates the price at a given tick index based on the provided
+// starting price of 1 at k=0. The price is calculated using a square root function
+// with a coefficient of 9, where the price increases by a factor of 10 for every
+// increment of k.
+//
+// The function takes in two arguments:
+// 	- tickIndex: the tick index to calculate the price for
+// 	- kAtPriceOne: the value of k at which the starting price of 1 is set
+//
+// It returns a sdk.Dec representing the calculated price and an error if any errors
+// occurred during the calculation.
+func TickToPrice(tickIndex, kAtPriceOne sdk.Int) (price sdk.Dec, err error) {
+	if tickIndex.IsZero() {
+		return sdk.OneDec(), nil
+	}
+
+	var kIncrementDistance sdk.Dec
+	// The formula is as follows: k_increment_distance = 9 * 10**(-k_at_price_1)
+	// Due to sdk.Power restrictions, if the resulting power is negative, we take 9 * (1/10**k_at_price_1)
+	if kAtPriceOne.GTE(sdk.ZeroInt()) {
+		kIncrementDistance = sdk.NewDec(9).Mul(sdk.OneDec().Quo(sdk.NewInt(10).ToDec().Power(kAtPriceOne.Uint64())))
 	} else {
-		sqrtPrice, err = sdk.OneDec().Quo(sdk.NewDecWithPrec(10001, 4).Power(tickIndex.Abs().Uint64())).ApproxSqrt()
+		kIncrementDistance = sdk.NewDec(9).Mul(sdk.NewInt(10).ToDec().Power(kAtPriceOne.Abs().Uint64()))
 	}
-	if err != nil {
-		return sdk.Dec{}, err
+
+	// Use floor division to determine how many k increments we have passed
+	kDelta := tickIndex.ToDec().Quo(kIncrementDistance).TruncateInt()
+
+	// Calculate the current k value from the starting k value and the k delta
+	curK := (kAtPriceOne.Add(kDelta))
+
+	var curIncrement sdk.Dec
+	if curK.GTE(sdk.ZeroInt()) {
+		curIncrement = sdk.NewDec(10).Power(curK.Uint64())
+	} else {
+		curIncrement = sdk.NewDec(1).Quo(sdk.NewDec(10).Power(curK.Abs().Uint64()))
 	}
```

Comment:
We can probably define a helper function for this since it seems that this check for the quotient sign repeated three times within this function